source/Engine.cpp

```diff
@@ -1394,6 +1394,13 @@ void Engine::EnterSystem()
 		Messages::Add(GameData::HelpMessage("basics 1"), Messages::Importance::High);
 		Messages::Add(GameData::HelpMessage("basics 2"), Messages::Importance::High);
 	}
+
+	// If the flagship is targetable when it enters a system,
+	// it has encountered all targetable ships in the system.
+	if(flagship->IsTargetable())
+		for(const shared_ptr<Ship> &ship : ships)
+			if(ship->GetSystem() == system && ship->IsTargetable() < 1.)
+				eventQueue.emplace_back(player.FlagshipPtr(), ship, ShipEvent::ENCOUNTER);
```

Comment:
I think this might be dead code now. EnterSystem is called the moment you enter the system, but a ship is never targetable the moment it enters the system; either its zoom is <1 because it's taking off, or it's still in the middle of hyperspace.

Reply:
Hm. That's unfortunate. Is there a place to put this code that's "right after system entry, once the ship is targetable"?

Reply:
I'm pretty sure the movement loop actually captures all cases now.